### ISSUE TITLE
Still no subscription confirmation

### ISSUE THREAD

**Issue by sowee15** [localstack/localstack#1708]
Hi, I'm reposting since I can't re-open, sorry if you saw my other message, didn't want to duplicate. I'm using the latest docker image, but my endpoint still doesn't get called with subscription confirmation. I'm using `http://host.docker.internal` as endpoint's hostname. However, it does get called when publishing. I did a `docker pull localstack/localstack` to get the latest version and restarted localstack. Anything I'm missing? Thanks!

_Originally posted by @sowee15 in https://github.com/localstack/localstack/issues/881#issuecomment-547214638_

**Comment by whummer** [comment]
Thanks for reporting @sowee15 . I just did a quick test and it seems to be working for me (this is on Mac OS):

Terminal 1:
```
SERVICES=sns localstack start
```
Terminal 2:
```
nc -lt 5001
```

Terminal 3:
```
awslocal sns subscribe --protocol http --topic-arn arn:aws:sns:eu-central-1:000000000000:t1 --notification-endpoint http://host.docker.internal:5001
```

... output on terminal 2:
```
POST / HTTP/1.1
Host: host.docker.internal:5001
User-Agent: python-requests/2.22.0
Accept-Encoding: gzip, deflate
Accept: */*
Connection: keep-alive
Content-Type: text/plain
x-amz-sns-message-type: SubscriptionConfirmation
Content-Length: 521

{"MessageId": "6e31b006-41e3-45f1-aab7-940f6b5542ac", "Type": "SubscriptionConfirmation", "Timestamp": "2019-11-21T16:45:25.897886Z", "Message": "You have chosen to subscribe to the topic arn:aws:sns:eu-central-1:000000000000:t1.\nTo confirm the subscription, visit the SubscribeURL included in this message.", "TopicArn": "arn:aws:sns:eu-central-1:000000000000:t1", "Token": "04982e2d", "SubscribeURL": "http://localhost:4575/?Action=ConfirmSubscription&TopicArn=arn:aws:sns:eu-central-1:000000000000:t1&Token=04982e2d"}^C
```

**Comment by sowee15** [comment]
Hi,
Sorry for taking so long to get back. I just removed all my localstack images, pulled latest, restarted docker, ran the same test as you and it works. So I guess somewhow I wasn't running the latest images previously. Thanks for the follow up!

**Comment by sowee15** [comment]
Hi @whummer, 
In my project I still wasn't getting subscription confirmation calls, and I think I finally found what the issue is: in my project, I use attributes while subscribing. So I tried to subscribe while specifying an attribute in your test case and netcat is not outputting anything when `--attribute` is specified. For instance, 
```
awslocal sns subscribe --topic-arn arn:aws:sns:us-east-1:000000000000:t1 --protocol http --notification-endpoint http://host.docker.internal:5001/attributes-empty --attributes '{}'
``` 
or 
```
awslocal sns subscribe --topic-arn arn:aws:sns:us-east-1:000000000000:t1  --protocol http --notification-endpoint http://host.docker.internal:5001/attributes --attributes '{"FilterPolicy":"{\"actionType\":[{\"prefix\":\"test\"}]}"}'
``` 
doesn't seem to trigger subscription confirmation. Let me know if it's the same for you.

**Comment by whummer** [comment]
Thanks for the update @sowee15 - we were able to reproduce the issue based on the example you've provided. This should be fixed in #2639 - can you please give it a try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks
